lib/beaker/puppet_install_helper.rb

```diff
@@ -108,10 +108,23 @@ def find_install_version
     ENV['PUPPET_INSTALL_VERSION'] || ENV['PUPPET_VERSION']
   end
 
+  def hosts_contains_windows_agent?(hosts)
+    hosts_with_role(hosts, 'agent').select { |host| host['platform'].match(/^windows/) }.size > 0
+  end
+
   def install_agent_on(hosts, collection, version)
     if ENV['PUPPET_AGENT_SHA'].nil?
-      opts = options.merge(puppet_collection: collection,
-                           version: version)
+      opts = if hosts_contains_windows_agent?(hosts)
+               if version && version.match(/5\.\d\.\d/)
+                 options.merge(version: version,
+                               puppet_collection: 'puppet')
```

Comment:
we could also use 'puppet5' here, which i guess is what beaker-puppet is expecting, but 'puppet' works.